lib/matplotlib/image.py

```diff
@@ -1192,13 +1200,17 @@ def set_data(self, x, y, A):
 
         Parameters
         ----------
-        x, y : 1D array-likes or None
-            Monotonic arrays of shapes (N + 1,) and (M + 1,), respectively,
-            specifying rectangle boundaries.  If None, will default to
+        x, y : 1D array-like, optional
+            Monotonic arrays of length N+1 and M+1, respectively, specifying
```

Comment:
Stupid question: what happens in the arrays are not monotonic?

Reply:
I didn't check, but probably the drawing is messed up. I'd assume that the edges define poligons that are just rendered and will overlap if  the edges are not monotonic.